### ISSUE TITLE
Yutani: checking icon for NULL too late

### ISSUE THREAD

**Issue by turbocat2001** [klange/toaruos#276]
Yutani doesn't check icon for `NULL`, probably causing applications to crash when setting the caption like this: 
```c
SDL_WM_SetCaption("Wolfenstein 3D", NULL);
```

https://github.com/klange/toaruos/blob/abe66fb45ba60c26a927023706e25de34493ed98/lib/yutani.c#L809

**Comment by klange** [comment]
`yutani_window_advertise_icon` doesn't accept `NULL` for the icon. The bug lies in the SDL backend: `TOARU_SetCaption` should be choosing `yutani_window_advertise` if `icon` is `NULL`. I'll transfer this issue to the SDL fork.

**Comment by klange** [comment]
Ah, I don't have issues enabled for that repository... I'll do that later.

**Comment by klange** [comment]
I do see that there is a null check in `yutani_window_advertise_icon` but that is misleading...

**Comment by turbocat2001** [comment]
In that case, this if is redundant: https://github.com/klange/toaruos/blob/abe66fb45ba60c26a927023706e25de34493ed98/lib/yutani.c#L816

**Comment by klange** [comment]
Yeah. Maybe it _should_ support a `NULL` icon... and it should probably also support a `NULL` name, which it fails to do in the same way... :\

**Comment by turbocat2001** [comment]
Will you fix it? In my opinion, the fix is ​​quite obvious.

**Comment by klange** [comment]
I've got a fix pending on the SDL side. Still thinking about what I want to do here. Falling back to the no-icon form is likely the right choice, though I am also thinking about what should be done in both functions if the title is `NULL` - perhaps a signal to remove the window from the advertised window list.
